README.md

```diff
@@ -1,168 +1,156 @@
+```html
+<script type="speculationrules">
+{
+  "allow": [
+    {
+      "action": "prerender",
+      "url_patterns": ["/**"]
+    }
+  ],
+  "disallow": [
+    {
+      "action": "prerender",
+      "urls": ["/logout"]
+    }
+  ]
+}
+</script>
+```
 
-The origin may already have data in its unpartitioned storage, but from the above, our goal is to provide for prerendering which does not grant access to it until the user actually navigates to the destination site. However, this may cause the content to load and behave differently than it would have with access to its unpartitioned storage. Then, when the prerendered document is preseneted, the user will observe any such differences as brokenness.
+This indicates to the browser that all of the links it sees, except for any to `/logout`, are safe to prerender. The browser can then heuristically perform such prerendering when it has spare resources (bandwidth, CPU cycles, memory, ...). The browser could use any triggers it wanted for this heuristic prerendering, such as:
 
-For example, an authenticated user or subscriber will observe that on navigation, that state is not reflected (they appear logged out or encounter a paywall). The user is likely to __blame the innocent destination site for this brokenness__. (Naturally, this is unlikely to be an issue if the origin has no existing cookies or other storage.)
+* Historical data from the current user
+* Historical data aggregated over many users via telemetry
+* Behavior patterns for similar sites (e.g., often users click on one of the top N product listings/comments links)
+* Just-in-time behavior patterns (e.g., mouse hover)
 
-The user agent could reload the page on navigation, but this would defeat the point of prerendering.
+This can be supplemented via per-page tweaks to increase the strength of the suggestion. For example, the `<script>` block could have
 
-If the user agent could detect brokenness, it could avoid this. However, this seems impossible to do in general, because:
-* the server response could depend on the presence of cookies and other credentials
-* client-side access can be detected, but it is unclear how to infer whether this corresponds to user-visible brokenness
+```json
+{
+  "action": "prerender",
+  "selectors": [".high-likelihood-prerender"],
+  "likelihood": "high"
+}
+```
 
-It might be possible for the user agent to augment author declarations with a list of origins or documents known to behave well, as a browser feature. This would depend on identifying such sites in a fairly reliable way and providing a mechanism for users to reload if they observe brokenness. However, such a mechanism would have limitations and would necessarily not behave the same in all browsers. The web platform should provide a way to more predictably get the desired behavior.
+and then decorate certain `<a>` elements with `class="high-likelihood-prerender"`.
```

Comment:
Noting, of course, that it would be totally legitimate to split off selector support into a followup, if we're worried about it. (My doc describes it because I do think it might be useful, and because it demonstrates extensibility.)

Reply:
Yeah, in general I'm not sure about these "scenarios" sections. They're currently more detailed than the sub-documents they link to. Maybe when we have the full explainer, we can change this to reference a "potential feature", linking to an explanation of it as being an extension point.